pkg/controller/awsmachine/awsmachine_controller.go

```diff
@@ -69,18 +103,283 @@ type ReconcileAWSMachine struct {
+	m, err := r.getMachineOwner(ctx, awsm.ObjectMeta)
+	if err != nil {
+		return reconcile.Result{}, err
+	} else if m == nil {
+		klog.Infof("Waiting for Machine Controller to set OwnerRef on AWSMachine %q/%q", awsm.Namespace, awsm.Name)
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Make sure bootstrap data is available and populated.
+	if m.Spec.Bootstrap.Data == nil || *m.Spec.Bootstrap.Data == "" {
+		klog.Infof("Waiting for bootstrap data to be available on AWSMachine %q/%q", awsm.Namespace, awsm.Name)
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	scope, err := actuators.NewMachineScope(actuators.MachineScopeParams{
+		ProviderMachine: awsm,
+		Machine:         m,
+		Client:          r.Client,
+	})
+	if err != nil {
+		return reconcile.Result{}, errors.Errorf("failed to create scope: %+v", err)
+	}
+	defer scope.Close()
+
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileAWSMachine) reconcile(ctx context.Context, scope *actuators.MachineScope) error {
+	exist, err := r.exists(scope)
+	if err != nil {
+		klog.Errorf("Failed to check if Machine %q infrastructure exists: %v", scope.Name(), err)
+		return err
+	}
+
+	// Reconcile ProviderID.
+	if pid := scope.GetProviderID(); pid == nil || *pid == "" {
+		scope.SetProviderID(fmt.Sprintf("aws:////%s", *scope.GetInstanceID()))
+	}
+
+	if exist {
+		scope.Info("Reconciling Machine triggers idempotent update")
+		return r.update(scope)
+	}
+
+	// Machine resource created. Machine does not yet exist.
+	scope.Info("Reconciling Machine triggers idempotent create")
+	return r.create(scope)
+}
+
+// create creates a machine and is invoked by the machine controller.
+func (r *ReconcileAWSMachine) create(scope *actuators.MachineScope) error {
+	if scope.Parent.Cluster.Annotations[infrav1.AnnotationClusterInfrastructureReady] != infrav1.ValueReady {
+		scope.Info("Cluster infrastructure is not ready yet - requeuing machine")
+		return &capierrors.RequeueAfterError{RequeueAfter: waitForClusterInfrastructureReadyDuration}
+	}
+
+	ec2svc := ec2.NewService(scope.Parent)
+
+	scope.Info("Retrieving machines for cluster")
+	machineList := &clusterv1.MachineList{}
+	if err := r.List(context.Background(), machineList, actuators.ListOptionsForCluster(scope.Parent.Name())); err != nil {
+		return errors.Wrapf(err, "failed to retrieve machines in cluster %q", scope.Parent.Name())
+	}
+
+	controlPlaneMachines := util.GetControlPlaneMachinesFromList(machineList)
+	if len(controlPlaneMachines) == 0 {
+		scope.Info("No control plane machines exist yet - requeuing")
+		return &capierrors.RequeueAfterError{RequeueAfter: waitForControlPlaneMachineExistenceDuration}
+	}
+
+	// Create the Machine.
+	i, err := ec2svc.CreateOrGetMachine(scope)
+	if err != nil {
+		return errors.Errorf("failed to create or get machine: %+v", err)
+	}
+
+	scope.SetInstanceID(i.ID)
+	scope.SetInstanceState(i.State)
+	scope.SetAnnotation("cluster-api-provider-aws", "true")
```

Comment:
Shouldn't this annotation be namespaced?

Reply:
We can definitely namespace it, I copied it from the old codebase